File: Data Structures & Algorithms/two-integer-sum/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var valueToIndex = new Dictionary<int, int>();
        int index = 0;

        foreach (var num in nums) {
            int complement = target - num;

            if (valueToIndex.ContainsKey(complement)) {
                return new int[] { valueToIndex[complement], index };
            }

            // 為避免重複鍵，僅當尚未加入過才加入
            if (!valueToIndex.ContainsKey(num)) {
                valueToIndex[num] = index;
            }

            index++;
        }

        return null; // 題目保證一定有解
    }
}

File: Data Structures & Algorithms/subarray-sums-divisible-by-k/submission-0.java
class Solution {
    public int subarraysDivByK(int[] nums, int k) {
        int res = 0;
        for (int i = 0; i < nums.length; i++){
            int totalSum = 0;
            for (int j = i; j < nums.length; j++){
                totalSum += nums[j];
                if (totalSum % k == 0){
                    res += 1;
                }
            }
        }
        return res;
    }
}